### PR TITLE
Fixes #5316 : Add validation for is_event_online

### LINKED ISSUE #5286: Add support for online event
**Is your feature request related to a problem? Please describe.**

Support for online event needs to be added, like in EventBrite

**Describe the solution you'd like**

A property `is_event_online` needs to be added, and checks for location need to be handled to allow publishing for online events

### LINKED ISSUE #5316: Add validation for is_event_online
**Describe the bug**

Add validation for is_event_online
**To Reproduce**

**Expected behavior**

**Stacktrace**

**Additional details (please complete the following information):**
 - OS: [e.g. MacOS, Ubuntu, CentOS]
 - Python Version [e.g. `3.5`, `3.6`]
 - `HEAD` Commit hash [e.g. `4629c62`]

**Additional context**

**Wanna work on this issue**

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #5316 : Add validation for is_event_online

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Fixes #5316 : Add validation for is_event_online

#### Changes proposed in this pull request:

Fixes #5316 : Add validation for is_event_online

screenshot from 2018-08-07 18-09-25

### CODE REVIEW

**Comment by mishuvs** [comment]
@bhaveshAn I had spent a day working on that. I was going to send a PR after finalizing. When I asked you on the issue, you didn't reply and opened your own PR.

**Comment by bhaveshAn** [comment]
> When I asked you on the issue, you didn't reply

Asked me the implementation and not willingness to work.

**Comment by sridharjajoo** [comment]
@bhaveshAn Kindly make sure that when an event is online then the event can get published without entering the location details

File: app/api/events.py [hunk 1]
```diff
@@ -121,10 +121,19 @@ def before_post(self, args, kwargs, data=None):
...
                                      "Only verified accounts can publish events")
-        if data.get('state', None) == 'published' and not data.get('location_name', None):
+        if data.get('state', None) == 'published' and not data.get('location_name', None) and \
+                not data.get('is_online'):
             raise ConflictException({'pointer': '/data/attributes/location-name'},
                                     "Location is required to publish the event")
 
+        if data.get('location_name', None) and data.get('is_online'):
+            raise ConflictException({'pointer': '/data/attributes/location-name'},
+                                    "Online Event does not have any locaton")
```
Comment (sridharjajoo) [inline app/api/events.py]: I will eventually hide the location fields in orga app in case of online events so you may as well remove this

Comment (bhaveshAn) [reply app/api/events.py]: This is not particular to orga-app even a user who is creating event directly using APIs should also mind this thing.

File: app/models/event.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ class Event(SoftDeletionModel):
     starts_at = db.Column(db.DateTime(timezone=True), nullable=False)
     ends_at = db.Column(db.DateTime(timezone=True), nullable=False)
     timezone = db.Column(db.String, nullable=False, default="UTC")
-    is_event_online = db.Column(db.Boolean)
+    is_event_online = db.Column(db.Boolean, default=False)
```
Comment (srv-twry) [inline app/models/event.py]: This will require you to run the migrations. Right ?

Comment (bhaveshAn) [reply app/models/event.py]: Migrations add default values as specified in models only when the attr is required. 
Here it's not mentioned that it's value is required in the model or schema.

**Pull request merged**